src/pg/sql/32_knn_prediction.sql

```diff
@@ -0,0 +1,46 @@
+
+CREATE OR REPLACE FUNCTION CDB_knnWeightedAverage(source_geom geometry(Point, 4326), target_geoms geometry(Point, 4326)[], target_vals numeric[], num_neighbors INT)
+RETURNS numeric
+AS $$
+DECLARE
+  weighted_avg numeric;
+  vals numeric[];
+  distances numeric[];
+  idx INT;
+BEGIN
+
+  IF array_length(target_geoms) IS NULL
+  THEN
+    RETURN NULL;
+  END IF;
+
+  EXECUTE
+  'SELECT
+     array_agg(vals),
+     array_agg(ST_Distance($1::geography, geoms::geography)) As dist
+   FROM (
+     SELECT geoms, vals FROM (
+       SELECT unnest($2) As geoms, unnest($3) As vals
+     ) As i
+     ORDER BY $1 <-> geoms
+     LIMIT $4) As j'
+  USING source_geom, target_geoms, target_vals, num_neighbors
+  INTO vals, distances;
+
+  IF 0 = Any(distances)
```

Comment:
You might be able to avoid that first check by using COALESCE in the sum bellow 

so 
```postgresql
weighted_avg := (SELECT sum(COALESCE(v / d, v)) / sum(COALESCE(1.0 / d, 1)) FROM (SELECT unnest(vals) As v, unnest(distances) As d) As x);
```
then if d is 0 then the sum is equal to v ?